Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,384 @@
+
+      latest_s =
+        if info["version"]["outdated"]
+          "#{Tty.green}#{info["version"]["latest"]}#{Tty.reset}"
+        else
+          info["version"]["latest"]
+        end
+
+      puts "#{formula_s} : #{current_s} ==> #{latest_s}"
+    end
+
+    def checkable_urls(formula)
+      urls = []
+      urls << formula.head.url if formula.head
+      if formula.stable
+        urls << formula.stable.url
+        urls.concat(formula.stable.mirrors)
+      end
+      urls << formula.homepage if formula.homepage
+
+      urls.compact
+    end
+
+    def preprocess_url(url)
+      # Check for GitHub repos on github.com, not AWS
+      url.sub!("github.s3.amazonaws.com", "github.com") if url.include?("github")
+
+      # Use repo from GitHub or GitLab inferred from download URL
+      if url.include?("github.com") && GITHUB_SPECIAL_CASES.none? { |sc| url.include? sc }
+        if url.include? "archive"
+          url = url.sub(%r{/archive/.*}, ".git") if url.include? "github"
+        elsif url.include? "releases"
+          url = url.sub(%r{/releases/.*}, ".git")
+        elsif url.include? "downloads"
+          url = Pathname.new(url.sub(%r{/downloads(.*)}, "\\1")).dirname.to_s+".git"
+        elsif !url.end_with?(".git")
+          # Truncate the URL at the user/repo part, if possible
+          github_repo_url = url[%r{((?:[a-z]+://)?github.com/[^/]+/[^/#]+)}]
+          url = github_repo_url unless github_repo_url.blank?
+
+          url = url[0..-2] if url.end_with?("/")
+
+          url += ".git"
+        end
+      elsif url.include?("/-/archive/")
+        url = url.sub(%r{/-/archive/.*$}i, ".git")
+      end
+
+      url
+    end
+
+    def latest_version(formula, args:)
+      has_livecheckable = formula.livecheckable?
+      livecheck = formula.livecheck
+      livecheck_regex = livecheck.regex
+      livecheck_strategy = livecheck.strategy
+      livecheck_url = livecheck.url
+
+      urls = [livecheck_url] if livecheck_url.is_a?(String) && !livecheck_url.blank?
+      urls ||= checkable_urls(formula)
+
+      if args.debug?
+        puts "\nFormula:          #{formula_name(formula, args: args)}"
+        puts "Head only?:       #{!formula.stable?}" unless formula.stable?
+        puts "Livecheckable?:   #{has_livecheckable ? "Yes" : "No"}"
+      end
+
+      urls.each_with_index do |original_url, i|
+        puts "\nURL:              #{original_url}" if args.debug?
+
+        # Skip Gists until/unless we create a method of identifying revisions
+        if original_url.include?("gist.github.com")
+          odebug "Skipping: GitHub Gists are not supported"
+          next
+        end
+
+        # Do not preprocess the URL when livecheck.strategy is set to :page_match
+        url = (livecheck_strategy == :page_match) ? original_url : preprocess_url(original_url)
+
+        strategies = Strategy.from_url(url, livecheck_regex.present?)
+        strategy = Strategy.from_symbol(livecheck_strategy) || strategies[0]
```

Comment:
```suggestion
        strategy = Strategy.from_symbol(livecheck_strategy) || strategies.first
```

although I think

```suggestion
        strategy = Strategy.from_symbol(livecheck_strategy)
        strategy ||= strategies.first
```

is often easier to follow in these cases.